pkg/operation/botanist/component/resourcemanager/resource_manager.go

```diff
@@ -882,15 +891,53 @@ func (r *resourceManager) ensureShootResources(ctx context.Context) error {
 				Namespace: metav1.NamespaceSystem,
 			}},
 		}
+
+		failurePolicy = admissionregistrationv1.Ignore
+		matchPolicy   = admissionregistrationv1.Exact
+		sideEffect    = admissionregistrationv1.SideEffectClassNone
+
+		podSchedulerNameMutatingWebhookConfiguration = &admissionregistrationv1.MutatingWebhookConfiguration{
```

Comment:
Why are you creating a dedicated `MutatingWebhookConfiguration` for this? We already have https://github.com/gardener/gardener/blob/715966f8767f1d3516fc435fac1452385087cfd5/pkg/operation/botanist/component/resourcemanager/resource_manager.go#L913 and you could simply extend this so that we only have one webhook resource for GRM in the shoots? (same applies to your YAMLs in the `example` directory)

Reply:
Yes, I was also wondering about this when I was doing the changes. I guess both of the options are fine for me. Maybe I did it via separate webhook because the 2 existing webhooks were more coupled to GRM while this one is not. Should I add it to the existing `MutatingWebhookConfiguration`?

Reply:
Do you mean that the webhook could theoretically be served by another binary? Probably the same applies to the existing webhooks as well. Generally, given that it's now part of GRM and that there doesn't seem to be a reason for having it separately, I would vote for adding it to the existing `MutatingWebhookConfiguration` (everything in one place) 👍🏻

Reply:
I now added it in the existing `MutatingWebhookConfiguration`.